src/parser.js

```diff
@@ -1189,8 +1189,11 @@ var Parser = Object.extend({
 
                 // Same for the succeding block start token
                 if(nextToken &&
-                   nextToken.type === lexer.TOKEN_BLOCK_START &&
-                   nextVal.charAt(nextVal.length - 1) === '-') {
+                    ((nextToken.type === lexer.TOKEN_BLOCK_START &&
+                      nextVal.charAt(nextVal.length - 1) === '-') ||
+                    (nextToken.type === lexer.TOKEN_COMMENT &&
+                      nextVal.charAt(this.tokens.tags.COMMENT_START.length)
```

Comment:
Because the `TOKEN_COMMENT` contains `COMMENT_START` and `COMMENT_END`. It's a whole token so that the `nextVal` here is not `{%`.  

Maybe breaking the `TOKEN_COMMENT` into three parts would be better. But It need change more code so that I use the tag instead.